java/opendcs/src/main/java/decodes/tsdb/algo/AW_AlgorithmBase.java

```diff
@@ -1653,7 +1655,36 @@ protected PropertySpec[] getAlgoPropertySpecs()
 									final org.opendcs.annotations.PropertySpec propSpec = p.second;
 									final String name = PropertySpec.getPropertyName(f, propSpec);
 									String specType = PropertySpec.getSpecTypeFromAnnotation(propSpec, f);
-									return new PropertySpec(name, specType, propSpec.description());
+									
+									// Create PropertySpec
+									PropertySpec spec = new PropertySpec(name, specType, propSpec.description());
+									
+									// Handle requirement groups
+									org.opendcs.annotations.PropertySpec.RequirementGroupDef[] groupDefs = propSpec.requirementGroups();
```

Comment:
This should definitely be a utility, RoutingSpecs, DataSource, just about everything will be using these annotations.